### ISSUE TITLE
AvoidEscapedUnicodeCharactersCheck: False positive on text blocks with a line separation

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#9159]
Taken from https://github.com/checkstyle/checkstyle/pull/9135 ,

````
$ cat TestClass.java
public class TestClass {
    void method() {
          String allCharactersEscaped = """
              \u03bc\u03bc""";
          String allCharactersEscaped2 = "\n\u03bc\u03bc";
          String allCharactersEscaped3 = """
              \u03bc\u03bc
              \u03bc\u03bc
              """;
          String allCharactersEscaped2 = "\n\u03bc\u03bc\n\u03bc\u03bc\n";
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="AvoidEscapedUnicodeCharacters">
  <property name="allowIfAllCharactersEscaped" value="true"/>
</module>
    </module>
</module>

$ java -jar checkstyle-8.39-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:3:44: Unicode escape(s) usage should be avoided. [AvoidEscapedUnicodeCharacters]
[ERROR] TestClass.java:6:45: Unicode escape(s) usage should be avoided. [AvoidEscapedUnicodeCharacters]
Audit done.
Checkstyle ends with 2 errors.
````

Line 3 and 6 is a false positive and should not report a violation as the leading blank spaces in the text block after the new lines should basically be ignored.

**Comment by nmancus1** [comment]
This is not entirely accurate; it is the leading whitespace (indent) causing the false positives.

>Also, without stripping the text block string of the indent/initial newline, check was giving false positives for strings that were actually all unicode/escape chars.

This sentence might have given the wrong idea; I grouped the "indent/ initial newline" together since that is the functionality of the `CheckUtil.stripIndentAndInitialNewLineFromTextBlock()` method, sorry for any confusion.

**Comment by rnveach** [comment]
I updated the sentence. If you it is still an issue, please provide sentence you would prefer.

**Comment by nmancus1** [comment]
This is correct, please approve.

**Comment by rnveach** [comment]
Fix was merged
